plugins/lua/buildingplan/planneroverlay.lua

```diff
@@ -452,6 +456,101 @@ local function get_placement_errors()
     return out
 end
 
+--------------------------------
+-- QuickFilter
+--
+
+-- Used to store a table of the following format:
+-- table<integer, { label: string, mats: string[] }>
+-- integer: quick filter slot
+-- label: string representation of the filter
+-- mats: list of material names allowed by the filter
+BUILDINGPLAN_FILTERS_KEY = "buildingplan/quick-filters"
+
+QuickFilter = defclass(QuickFilter, widgets.Panel)
+QuickFilter.ATTRS{
+    idx=DEFAULT_NIL,
+    on_click_fn=DEFAULT_NIL,
+    is_selected_fn=function () return false end,
+}
+
+function QuickFilter:init()
+    local label = ('%d.'):format(self.idx)
+    self.frame.w = 27
+    self.renaming = false
+
+    self:addviews {
+        widgets.Label{
+            frame={t=0, l=0},
+            text=string.char(16), -- this is the "►" character
+            text_pen=COLOR_YELLOW,
+            auto_width=true,
+            visible=self.is_selected_fn,
+        },
+        widgets.Label { frame = { t = 0, l = 2 }, text = label },
+        widgets.Label {
+            frame = { t = 0, l = 5 },
+            text = { { text = self:callback('get_label_text'), pen = function() return COLOR_CYAN end } },
+            width = item_filter_chars + 2,
+            visible = function () return self.renaming == false end,
+            on_click = self:callback("on_click"),
+        },
+        widgets.EditField {
+            view_id='edit_field',
+            frame = { t = 0, l = 5 },
+            text = "",
+            width = item_filter_chars + 2,
```

Comment:
`width` isn't an attribute for `EditField`, is it (or `Label` above)? should that instead be the `w` key in the frame or, for the `Label` at least, the `width` key in the text token?